src/widgets/dnshole/DnsHoleControls.tsx

```diff
@@ -124,10 +142,17 @@ function DnsHoleControlsWidgetTile({ widget }: DnsHoleControlsWidgetProps) {
     return dnsList;
   };
 
-  const toggleDns = async (action: 'enable' | 'disable', appsToChange?: string[]) => {
+  const toggleDns = async (
+    action: 'enable' | 'disable',
+    appsToChange?: string[],
+    hours: number = 0,
+    minutes: number = 0
+  ) => {
+    const duration = hours * 3600 + minutes * 60;
     await mutateAsync(
       {
         action,
+        duration,
         configName,
         appsToChange,
       },
```

Comment:
looks like this wont work anymore as well... assumption was that summary status only updated when toggleDns was called. now the summary status can be out of sync when the service is automatically renabled but summary data hasnt been refreshed

not sure the cleanest fix

Reply:
maybe this?

```
        onSettled: () => {
          trpcUtils.dnsHole.summary.invalidate();

          setTimeout(() => {
            trpcUtils.dnsHole.summary.invalidate();
          }, (duration + 3) * 1000)
        },
```

Reply:
setTimeout fix the problem. I made test on the old widget, the summary refresh don't work.